src/Components/Projects/Projects.tsx

```diff
@@ -151,31 +149,22 @@ const Projects: React.FC = () => {
         );
       });
   };
-  // **********search by proj name***********************
-  const getProjectTitleValue = (e: MouseEvent) => {
-   
-    getAllProjectsList(1, e.target.value);
-    setSearchString(e.target.value);
-   
-  };
   // *****************************************************
+ 
   useEffect(() => {
-      getAllProjectsList(1,searchString);
+    const timerId = setTimeout(() => {
```

Comment:
No need for debouncing here, as this action only happens when the page first renders.
The current code is just adding a normal delay of 500ms to the calling of the API.

Debouncing is used for example in cases such as searching, because we normally have the searchString that is changing in every keystroke and this means that we will make a call to the API on every change.

Debouncing solves this problem by only sending a call to the API after the user stops typing for delay we specified.